pkg/apis/crd/v1alpha2/types.go

```diff
@@ -210,6 +210,11 @@ type EgressSpec struct {
 	AppliedTo AppliedTo `json:"appliedTo"`
 	// EgressIP specifies the SNAT IP address for the selected workloads.
 	EgressIP string `json:"egressIP"`
+	// ExternalIPPool specifies the IP Pool that the EgressIP should be allocated from.
+	// If it's set, the allocated EgressIP will be assigned to a Node specified by the Pool and it will failover to
+	// another Node if the assigned Node becomes unavailable.
+	// If it's set with EgressIP, the EgressIP must be in the IP Pool.
+	ExternalIPPool string `json:"externalIPPool"`
```

Comment:
Should we have an `EgressStatus` to report which IP has been assigned (e.g. for troubleshooting) to the Egress resource?

Reply:
BTW, for troubleshooting and visibility, there is a plan to add a `nodeName` field to `status` after the major function is done.

Reply:
Sounds good. If we want to get fancy, we can even add a status to the EgressIPPool to report the number of allocated vs free IPs?

Reply:
Good suggestion, will create an issue to track it.